### PR TITLE
fix v10 CI tests

### PR DESCRIPTION
**Pull request by blagoev**
fix CI tests for v10

### CODE REVIEW

**Review by kneth — APPROVED** [review]

File: src/RealmJS.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -1226,9 +1235,16 @@
...
 				DEBUG_INFORMATION_FORMAT = dwarf;
 				HEADER_SEARCH_PATHS = (
-					"$(SRCROOT)/object-store/src",
-					"$(SRCROOT)/../vendor",
-					"$(SRCROOT)/../vendor/core/include",
+					"$(SRCROOT)/object-store/src/**",
+					"$(SRCROOT)/../vendor/**",
+					"$(SRCROOT)/../vendor/core/include/**",
+				);
+				"HEADER_SEARCH_PATHS[arch=*]" = (
```
Comment (kraenhansen) [inline src/RealmJS.xcodeproj/project.pbxproj:1242]: I wonder if this is the same as `HEADER_SEARCH_PATHS`?

Comment (blagoev) [reply src/RealmJS.xcodeproj/project.pbxproj:1242]: I don't know

**Comment by steffenagger** [comment]
I checked out this branch, and cleaned build folders etc.
Then executed: `(cd scripts && sh test.sh react-tests)`

But I'm getting:
```sh
In file included from /[local-path]/realm/js/tests/react-test-app/node_modules/realm/src/object-store/src/sync/push_client.cpp:19:
In file included from object-store/src/sync/push_client.hpp:22:
In file included from object-store/src/sync/auth_request_client.hpp:21:
object-store/src/sync/generic_network_transport.hpp:23:10: fatal error: 'realm/util/to_string.hpp' file not found
#include <realm/util/to_string.hpp>
```

Am I missing a step, or doing something wrong?

**Comment by kraenhansen** [comment]
> Am I missing a step, or doing something wrong?

Did you `git submodule update --recursive`?

**Comment by steffenagger** [comment]
> Did you `git submodule update --recursive`?

I hadn't, but same result.

File: tests/js/encryption-tests.js [hunk 2]
```diff
@@ -74,10 +81,15 @@ module.exports = {
     },
 
     testEncryptionWithSync: function() {
+        //TODO: remove when MongoDB Realm test server can be hosted on Mac or other options exists
+        if (!isNodeProcess) {
+            return Promise.resolve();;
+        }
```
Comment (kraenhansen) [inline tests/js/encryption-tests.js:87]: This should probably be listed on https://github.com/realm/realm-js/issues/3312

File: tests/js/index.js [hunk 3]
```diff
@@ -43,18 +43,25 @@ var TESTS = {
...
+//TODO: remove when MongoDB Realm test server can be hosted on Mac or other options exists
+if (isNodeProcess) {
+    TESTS.ObjectIDTests = require('./object-id-tests');
+}
+
 // If sync is enabled, run the sync tests
 if (global.enableSyncTests) {
-    TESTS.AppTests = require('./app-tests');
+    //TODO: remove when MongoDB Realm test server can be hosted on Mac or other options exists
+    if (isNodeProcess) {
```
Comment (kraenhansen) [inline tests/js/index.js:59]: This should probably be listed on #3312

File: tests/js/index.js [hunk 4]
```diff
@@ -43,18 +43,25 @@ var TESTS = {
     QueryTests: require('./query-tests'),
     MigrationTests: require('./migration-tests'),
     EncryptionTests: require('./encryption-tests'),
-    ObjectIDTests: require('./object-id-tests'),
     AliasTests: require('./alias-tests'),
     BsonTests: require('./bson-tests'),
     // Garbagecollectiontests: require('./garbage-collection'),
 };
 
+//TODO: remove when MongoDB Realm test server can be hosted on Mac or other options exists
```
Comment (kraenhansen) [inline tests/js/index.js:51]: This should probably be listed on #3312

File: tests/js/realm-tests.js [hunk 5]
```diff
@@ -18,11 +18,8 @@
...
 
-/* global navigator, WorkerNavigator */
-
-const require_method = require;
-
 // Prevent React Native packager from seeing modules required with this
+const require_method = require;
 function nodeRequire(module) {
     return require_method(module);
 }
```
Comment (kraenhansen) [inline tests/js/realm-tests.js:25]:
```suggestion
const nodeRequire = require;
```

File: tests/js/encryption-tests.js [hunk 6]
```diff
@@ -24,6 +24,13 @@ var Schemas = require('./schemas');
 
 const { ObjectId } = require("bson");
 
+// Prevent React Native packager from seeing modules required with this
+const require_method = require;
+function nodeRequire(module) {
+    return require_method(module);
+}
```
Comment (kraenhansen) [inline tests/js/encryption-tests.js:31]:
```suggestion
const nodeRequire = require;
```

File: tests/js/realm-tests.js [hunk 7]
```diff
@@ -448,6 +445,10 @@ module.exports = {
     },
 
     testRealmExists: function() {
+        //TODO: remove when MongoDB Realm test server can be hosted on Mac or other options exists
+        if (!isNodeProcess) {
+            return Promise.resolve();
+        }
```
Comment (kraenhansen) [inline tests/js/realm-tests.js:451]: This should probably be listed on #3312

File: tests/js/realm-tests.js [hunk 8]
```diff
@@ -1579,10 +1580,15 @@ module.exports = {
...
         if (!global.enableSyncTests) {
-            return;
+            return Promise.resolve();
         }
 
-        const appConfig = require('./support/testConfig').integrationAppConfig;
+        //TODO: remove when MongoDB Realm test server can be hosted on Mac or other options exists
+        if (!isNodeProcess) {
+            return Promise.resolve();
+        }
```
Comment (kraenhansen) [inline tests/js/realm-tests.js:1589]: This should probably be listed on #3312

File: tests/js/realm-tests.js [hunk 9]
```diff
@@ -1609,7 +1615,8 @@ module.exports = {
             });
     },
 
-    testNoMigrationOnSync: function() {
+    //TODO: enable when v10 CI is green
+    /*testNoMigrationOnSync: function() {
```
Comment (kraenhansen) [inline tests/js/realm-tests.js:1619]: This should probably be listed on #3312

File: tests/js/support/testConfig.js [hunk 10]
```diff
@@ -21,7 +21,24 @@
 
 // If the docker instance has imported this stitch config, it will have written the app id
 // back into the config file, so we can read it out again here.
-const integrationTestsAppId = `${require("../../../src/object-store/tests/mongodb/stitch.json").app_id}`;
+
+// Prevent React Native packager from seeing modules required with this
+const require_method = require;
+function nodeRequire(module) {
+    return require_method(module);
+}
```
Comment (kraenhansen) [inline tests/js/support/testConfig.js:29]:
```suggestion
const nodeRequire = require;
```

File: tests/js/support/testConfig.js [hunk 11]
```diff
@@ -21,7 +21,24 @@
...
+
+let pathToStitchJson = "../../../src/object-store/tests/mongodb/stitch.json";
+const isNodeProcess = typeof process === 'object' && process + '' === '[object process]';
+
+if (isNodeProcess && process.env.ELECTRON_TESTS_REALM_MODULE_PATH) {
+    const path = nodeRequire("path");
+    console.log("ELECTRON_TESTS_REALM_MODULE_PATH " + process.env.ELECTRON_TESTS_REALM_MODULE_PATH);
+    pathToStitchJson = path.resolve(process.env.ELECTRON_TESTS_REALM_MODULE_PATH, '../../../../src/object-store/tests/mongodb/stitch.json')
+}
+console.log("pathToStitchJson " + pathToStitchJson);
```
Comment (kraenhansen) [inline tests/js/support/testConfig.js:39]: Looks like debugging which should to be removed.

File: tests/js/support/testConfig.js [hunk 12]
```diff
@@ -21,7 +21,24 @@
...
+
+const integrationTestsAppId = `${nodeRequire(pathToStitchJson).app_id}`;
```
Comment (kraenhansen) [inline tests/js/support/testConfig.js:41]: Not big fan og calling functions from a string template (as errors thrown are a bit less obvious to debug from the stacktrace). Also the string template seems unnecessary.

```suggestion
const stitchConfiguration = nodeRequire(pathToStitchJson);
const integrationTestsAppId = stitchConfiguration.app_id;
```

File: tests/js/support/testConfig.js [hunk 13]
```diff
@@ -21,7 +21,24 @@
...
+function nodeRequire(module) {
+    return require_method(module);
+}
+
+let pathToStitchJson = "../../../src/object-store/tests/mongodb/stitch.json";
+const isNodeProcess = typeof process === 'object' && process + '' === '[object process]';
+
+if (isNodeProcess && process.env.ELECTRON_TESTS_REALM_MODULE_PATH) {
+    const path = nodeRequire("path");
+    console.log("ELECTRON_TESTS_REALM_MODULE_PATH " + process.env.ELECTRON_TESTS_REALM_MODULE_PATH);
```
Comment (kraenhansen) [inline tests/js/support/testConfig.js:36]: Looks like debugging which should to be removed.

**Review by kraenhansen — CHANGES_REQUESTED** [review]
A few places that I think we should mention in #3312

File: tests/electron/package.json [hunk 14]
```diff
@@ -5,13 +5,13 @@
   "main": "index.js",
   "scripts": {
     "start": "electron ./app/main.js",
-    "test": "node ./runner.js --js-flags=\"--expose-gc\""
+    "start-test": "electron --js-flags=\"--expose-gc\" --no-sandbox --enable-logging ./app/main.js"
```
[the remark this answers is not in the document]
Comment (blagoev) [reply tests/electron/package.json:8]: yes, will fix

File: tests/electron/package.json [hunk 15]
```diff
@@ -5,13 +5,13 @@
...
   },
   "devDependencies": {
-    "electron": "^8.4.1",
+    "electron": "^9.3.2",
     "jasmine": "^3.4.0",
     "jasmine-console-reporter": "^3.1.0",
-    "spectron": "^10.0.1"
+    "spectron": "^11.0.0"
```
[the remark this answers is not in the document]
Comment (blagoev) [reply tests/electron/package.json:14]: yes its not

File: tests/js/encryption-tests.js (hunk 6 above)
Comment (blagoev) [reply tests/js/encryption-tests.js:31]: this is copy paste from somewhere else

Comment (blagoev) [reply tests/js/encryption-tests.js:31]: the suggested change actually wont work

File: tests/js/encryption-tests.js (hunk 2 above)
Comment (blagoev) [reply tests/js/encryption-tests.js:87]: done

File: tests/js/index.js (hunk 4 above)
Comment (blagoev) [reply tests/js/index.js:51]: done

File: tests/js/index.js (hunk 3 above)
Comment (blagoev) [reply tests/js/index.js:59]: done

File: tests/js/realm-tests.js (hunk 7 above)
Comment (blagoev) [reply tests/js/realm-tests.js:451]: done

File: tests/js/realm-tests.js (hunk 5 above)
Comment (blagoev) [reply tests/js/realm-tests.js:25]: this wont work

File: tests/js/realm-tests.js (hunk 8 above)
Comment (blagoev) [reply tests/js/realm-tests.js:1589]: done

File: tests/js/realm-tests.js (hunk 9 above)
Comment (blagoev) [reply tests/js/realm-tests.js:1619]: done

File: tests/js/support/testConfig.js (hunk 10 above)
Comment (blagoev) [reply tests/js/support/testConfig.js:29]: wont work

File: tests/js/support/testConfig.js (hunk 11 above)
Comment (blagoev) [reply tests/js/support/testConfig.js:39]: not its not

File: tests/js/support/testConfig.js (hunk 12 above)
Comment (blagoev) [reply tests/js/support/testConfig.js:41]: not my change. see previous version

File: tests/js/support/testConfig.js (hunk 13 above)
Comment (blagoev) [reply tests/js/support/testConfig.js:36]: nope its not

File: tests/js/realm-tests.js (hunk 9 above)
Comment (blagoev) [reply tests/js/realm-tests.js:1619]: already was there

File: tests/js/support/testConfig.js (hunk 11 above)
Comment (blagoev) [reply tests/js/support/testConfig.js:39]: also don't agree debugging output should be removed from test infrastructure. When once it was found that without it its much harder to find a problem.

File: tests/js/realm-tests.js (hunk 5 above)
Comment (kraenhansen) [reply tests/js/realm-tests.js:25]: Why not? I believe I've used that elsewhere and it looks equivalent to me. Is it picked up by the RN bundler?

**Review by kraenhansen — APPROVED** [review]
I don't see why the suggested change to `const nodeRequire = require` won't work, but we can iterate on that later.

**Review by cesarvr — APPROVED** [review]

**Comment by blagoev** [comment]
this won't be able to hide it from the analyzer. Check how we are doing that on every other place.

File: tests/react-test-app/ios/ReactTests.xcodeproj/project.pbxproj [hunk 16]
```diff
@@ -450,6 +453,7 @@
 					"\"${PODS_ROOT}/Headers/Public/yoga\"",
 				);
 				INFOPLIST_FILE = ../../ios/Info.plist;
+				IPHONEOS_DEPLOYMENT_TARGET = 12.0;
```
Comment (kneth) [inline tests/react-test-app/ios/ReactTests.xcodeproj/project.pbxproj:456]: I believe that RN is supporting iOS 9.0

**Review by kneth — APPROVED** [review]
LGTM. My single comment is more a question.

**Comment by blagoev** [comment]
merging this one

**Pull request merged**